pandas/core/indexes/base.py

```diff
@@ -4866,6 +4867,20 @@ def slice_locs(self, start=None, end=None, step=None, kind=None):
             # If it's a reverse slice, temporarily swap bounds.
             start, end = end, start
 
+        # GH 16785: If start and end happen to be date strings with UTC offsets
+        # attempt to parse and check that the offsets are the same
+        if (isinstance(start, compat.string_types) and
+                isinstance(end, compat.string_types)):
+            try:
```

Comment:
should these not be individually converted?

Reply:
They are eventually, but when they are individually converted (after 3 nested function calls), it's impossible to infer what the UTC offset was in the original strings.

This is the earliest opportunity to check that the UTC offsets are the same.